packages/graphql-modules/src/application/apollo.ts

```diff
@@ -17,109 +17,58 @@ export interface ApolloRequestContext {
   };
 }
 
-export function apolloExecutorCreator({
-  createExecution,
-}: {
-  createExecution: Application['createExecution'];
-}): Application['createApolloExecutor'] {
-  return function createApolloExecutor(options) {
-    const executor = createExecution(options);
-    return async function executorAdapter(
-      requestContext: ApolloRequestContext
-    ) {
-      return executor({
-        schema: requestContext.schema,
-        document: requestContext.document,
-        operationName: requestContext.operationName,
-        variableValues: requestContext.request.variables,
-        contextValue: requestContext.context,
-      });
-    };
-  };
+export interface ApolloGatewayLoadResult {
+  executor: ApolloExecutor;
 }
 
-export function apolloSchemaCreator({
-  createSubscription,
-  contextBuilder,
-  schema,
-}: {
-  createSubscription: Application['createSubscription'];
-  contextBuilder: ExecutionContextBuilder;
-  schema: GraphQLSchema;
-}) {
-  const createApolloSchema = () => {
-    const sessions: Record<
-      string,
-      {
-        count: number;
-        session: {
-          destroy(): void;
-          context: InternalAppContext;
-        };
-      }
-    > = {};
-    const subscription = createSubscription();
+export type ApolloExecutor = (
+  requestContext: ApolloRequestContext
+) => Promise<ExecutionResult>;
 
-    function getSession(ctx: any) {
-      if (!ctx[CONTEXT_ID]) {
-        ctx[CONTEXT_ID] = uniqueId((id) => !sessions[id]);
-        const { context, ɵdestroy: destroy } = contextBuilder(ctx);
+export interface ApolloGatewayInterface {
+  onSchemaLoadOrUpdate(
+    callback: (schemaContext: {
+      apiSchema: GraphQLSchema;
+      coreSupergraphSdl: string;
+    }) => void
+  ): () => void;
+  load(): Promise<ApolloGatewayLoadResult>;
+  stop(): Promise<void>;
+}
 
-        sessions[ctx[CONTEXT_ID]] = {
-          count: 0,
-          session: {
-            context,
-            destroy() {
-              if (--sessions[ctx[CONTEXT_ID]].count === 0) {
-                destroy();
-                delete sessions[ctx[CONTEXT_ID]];
-                delete ctx[CONTEXT_ID];
-              }
-            },
+export function apolloGatewayCreator({
+  schema,
+  typeDefs,
+  createExecution,
+}: {
+  schema: Application['schema'];
+  typeDefs: Application['typeDefs'];
+  createExecution: Application['createExecution'];
+}): Application['createApolloGateway'] {
+  return function createApolloGateway(options) {
```

Comment:
@ardatan why do we have now `createApolloGateway`?

Reply:
Apollo removed `executor` so the only way of changing the executor is `gateway`, and it handles the schema as well.